common/ManagementTestShared/ManagementRecordTestBase.cs

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Resources;
+using System.Diagnostics;
+
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+
+namespace Azure.ResourceManager.TestFramework
+{
+    [Category("Recorded")]
+    public abstract class ManagementRecordTestBase<TEnvironment> : ClientTestBase where TEnvironment : TestEnvironment, new()
```

Comment:
Why don't you inherit from the RecordedTest base?

Reply:
Because we don't want to touch this file since we want it as a option to run the test as they currently are. Also, we don't want these [OneTimeSetup] [Setup] [OneTimeTearDown] [TearDown] to run in that file.

Reply:
I'm sorry I don't understand. What feature of RecordedTestBase would you like to avoid? You can make the base type [Setup]/etc method virtual and override them in you r class with any functionality you want.

Reply:
We could do that, but it is more logical to break these out to separate classes since they are truly doing separate things here, and move the common functionality into the class that they share, which is ClientBase.cs. This is more clear from a user tracing the code as well, since we wouldn't do anything with the setups, or teardownd, etc in the RecordedTestBase, we just be doing it to get some common functions from it, functions we could get by just moving them to a more common class, ClientTestBase.

Lastly, the semantics used in that class for Setup, TearDown, don't really match what we are now doing with the changes we are making. So overriding them won't make sense from a semantic point of view unless we changed the names to meet the semantics we want. We didn't really want to change this since this public functions are used in some places in the code and we don't want to break existing tests. Also, changing the semantics could confuse users who are calling these with trying to use them the current way.
Hence, it make sense to split this out for sematic reasons as well.

Reply:
>are truly doing separate things here,

What are these separate things? They both do test recording, the one you are adding just has more management specific features.

Reply:
I'll be supportive of renaming 4 existing base class method to the names you use in ManagementTestBase

Reply:
I like forking here for cleanness.  If we go the route you suggest @pakrym do you also suggest in the same PR we find / replace all usage / reference to these renames since that would be a breaking change on any Track 2 SDKs that already exist?

@bquantump can you do a quick search to see how many that would be just so we have the data?

Reply:
StartTestRecording is used 29 times in 14 files (I would count really 23 in 12 files since 6 are used in a coverage info and json)
StopTestRecording is used 20 time in 13 files (remove coverage files 16 times in 10 files)

Reply:
Are you counting non-source files?

Reply:
I don't see https://github.com/Azure/azure-sdk-for-net/search?q=StartTestRecording&type= 12 files here.

Reply:
Yes counting the coverage files, I assume this would have to change:

Example:

 <method name="StopTestRecording" signature="()" line-rate="0" branch-rate="0" complexity="2">

Reply:
Oh yes, the rest are coverage, so 12 files

Reply:
Can you list the files please? I see the search returning only 8.

Reply:
Updated

Reply:
I'd also like to better understand why this cannot inherit from RecordedTestBase? It seems like that would be the more natural thing to do instead of duplicating most of the RecordedTestBase class here.

Reply:
Our goal is not to duplicate and in fact I believe all we did was move the location of a few methods and NOT duplicate any of them

Reply:
It looks like you moved some recording based methods from the RecordedTestBase to ClientTestBase which is supposed too know anything about recordings. Instead we should just have this class inherit from RecordedTestBase and make tweaks there if needed.